client/src/pages/profile/index.tsx

```diff
@@ -0,0 +1,494 @@
+                      floating
+                      value={user.role}
+                    />
+                  </div>
+                ) : (
+                  <div className="input-field">
+                    <span>{`${parseCamelCase('role')}:`}</span>
+                    <Input
+                      value={convertToCapitalized(user.role)}
+                      transparent
+                      readOnly={!isEditMode}
+                    />
+                  </div>
+                )}
+                {isEditMode ? (
+                  <div className="input-field">
+                    <span>{parseCamelCase('dateJoined')}</span>
+                    <Input
+                      value={
+                        new Date(user.dateJoined).toISOString().split('T')[0]
+                      }
+                      readOnly={!isEditMode}
+                      transparent
+                      onChange={(e) =>
+                        setUser(
+                          updateUserField(
+                            user,
+                            'dateJoined',
+                            new Date(e.currentTarget.value),
+                          ),
+                        )
+                      }
+                      type="date"
+                      max={new Date(Date.now()).toISOString().split('T')[0]}
+                    />
+                  </div>
+                ) : (
+                  <div className="input-field">
+                    <span>{`${parseCamelCase('dateJoined')}:`}</span>
+                    <Input
+                      value={new Date(user.dateJoined).toLocaleDateString(
+                        'en-US',
+                        { timeZone: 'UTC' },
+                      )}
+                      transparent
+                      readOnly={!isEditMode}
+                    />
+                  </div>
+                )}
+              </Container>
+            </Grid.Column>
+            <Grid.Column textAlign="center" width={3}>
+              <HeaderTag as="h2">My Interests</HeaderTag>
+              {!isEditMode ? (
+                user.interests.map((interest, index) => (
+                  <div
+                    className={`field-label ${convertToClassName(interest)}`}
+                    key={index}
+                  >
+                    {convertToCapitalized(interest)}
+                  </div>
+                ))
+              ) : (
+                <Container style={{ paddingLeft: '20%' }}>
+                  {allInterests.map((interest, index) => (
+                    <div key={index} style={{ textAlign: 'left' }}>
+                      <Checkbox
+                        value={interest}
+                        label={convertToCapitalized(interest)}
+                        checked={user.interests.includes(interest)}
+                        onChange={(e, data) => editInterests(`${data.value!}`)}
+                      />
+                    </div>
```

Comment:
I think we should have the scroll feature for the interests list if there are too many to keep everything viewable in one screen without having to scroll on the overall page.

Reply:
Even if the user selects every interest possible it won't cause a scroll (or if they a smaller screen it might barely scroll a little). Im not sure how often people select every possible interest. The reason I like avoiding in page scrolling is because it results in random scroll bars that you can't really conditionally turn off and it can make the UI look kind of jank